scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1466,143 +1470,174 @@
 
     (.setHostnameAsFQDN pod-spec false)
 
-    ; main container
-    (let [{:keys [resources]} task-request
-          ;; NOTE: The scheduler's adjust-job-resources-for-pool-fn may modify :resources,
-          ;; whereas :scalar-requests always contains the unmodified job resource values.
-          {:keys [image port-mapping]} docker
-          ; gpu count is not stored in scalar-requests because Fenzo does not handle gpus in binpacking
-          gpus (or (:gpus resources) 0)
-          gpu-model-requested (constraints/job->gpu-model-requested gpus job pool-name)
-          ;; if disk config for pool has enable-constraint? set to true, add disk info to the job
-          enable-disk-constraint? (regexp-tools/match-based-on-pool-name (config/disk) pool-name :enable-constraint? :default-value false)
-          ;; if user did not specify disk request, use default on pool
-          disk-request (when enable-disk-constraint?
-                         (constraints/job-resources->disk-request resources pool-name))
-          disk-limit (when enable-disk-constraint? (-> resources :disk :limit))
-          ;; if user did not specify disk type, use default on pool
-          disk-type (when enable-disk-constraint? (constraints/job-resources->disk-type resources pool-name))
+    (let [{:keys [image]} docker
```

Comment:
Groktest: you needed to pull this up a level in order to use it in both the main container and the new checkpoint-init-container